querymsc.py

```diff
@@ -151,77 +151,56 @@ def query_annual_precip(self, session):
         Only the year from start and end times are used to create the
         time frame. All frequencies of observations are used, and the
         regular non-corrected air temperature is being used for this 
         calculation.
         -----------------------------------------
+        Args:
+            session (sqlalchemy Session): session constructed using
+                connection string and engine
+            percentile (float): desire percentile in fraction
         Returns: 
             query (sqlalchemy query): sqlalchemy query constructed 
                 using ORM to query temperature percentiles 
         """
 
-        # construct query table
+        # construct query
         query = (
-                session.query(
-                                (func.percentile_cont(0.01)
+                 session.query(
+                                (func.percentile_cont(percentile)
                                      .within_group(Obs.datum.asc())
                                      .label("temp")),
-                                 func.min(Obs.time).label("time_min"),
-                                 func.max(Obs.time).label("time_max"),
-                                 History.lat,
-                                 History.lon,
-                                 History.station_id)
-                            # select from and make join explicit
-                            .select_from([History, Variable])
-                            .filter(
-                                    and_(Obs.time >= self.start_time,
-                                         Obs.time <= self.end_time)
-                                    )
-                            .filter(
-                                    func.extract("month", Obs.time) == self.month
-                                    )
-                            .filter(
-                                    and_(Variable.standard_name == 'air_temperature',
-                                         Variable.id == 1510)
-                                    )
-                            .group_by(History.lat, 
-                                      History.lon, 
-                                      History.station_id)
+                                func.min(Obs.time).label("time_min"),
+                                func.max(Obs.time).label("time_max"),
+                                History.lat.label("lat"),
+                                History.lon.label("lon"),
+                                History.station_id.label("station_id"),
+                                func.count(Obs.datum).label('obs_count')
+                                 )
+                        .join(History)
+                        .join(Variable, Variable.id == Variable.id)
+                        .filter(
+                                and_(Obs.time >= self.start_time,
+                                     Obs.time <= self.end_time)
```

Comment:
You might want to consider using half-open intervals here, so that successive time ranges have no overlap whatsoever. Meaning make it `<` for the `end_time` comparison.